packages/router/addon/index.js

```diff
@@ -55,10 +63,22 @@ export default EmberRouter.extend({
       if (!bundle || bundle.loaded) {
         return original(name);
       }
-      return bundle.load().then(() => {
-        bundle.loaded = true;
-        return original(name);
-      });
+      this._inFlightLazyRoutes++;
+      return bundle.load().then(
```

Comment:
IMHO, we really want to leverage `ember-test-waiters` (what I've been working on to replace the `registerWaiter` system) which has a `waitForPromise` API that would fit this need precisely (and provide actual feedback in the console when delayed). At the moment that requires ember-test-helpers@1.6.0 (just released) and I haven't polyfilled it just yet.

```js
import { waitForPromise } from 'ember-test-waiters';

// ...snip...
return waitForPromise(
  bundle.load().then(() => {
    bundle.loaded = true;

    return original(name);
  })
);
```

[ember-test-waiters](https://github.com/rwjblue/ember-test-waiters)